src/utils/draftbots.js

```diff
@@ -215,268 +205,243 @@ export const getCastingProbability = (card, lands) => {
-    return (
-      getInternalSynergy(pickedInCombination, picked, cards) * getSynergyWeight(packNum, pickNum, initialState) +
-      getOpenness(combination, seen) * getOpennessWeight(packNum, pickNum, initialState) +
-      getColor(pickedInCombination, picked, probabilities, cards) * getColorWeight(packNum, pickNum, initialState) +
-      getFixing(combination, card, cards) * getFixingWeight(packNum, pickNum, initialState) +
-      (ratingScore + pickSynergyScore) * cardCastingProbability
-    );
-  }
-  // We can't filter on castable because that leads to returning
-  // -Infinity for changes so we see a flat hill and stop climbing.
-  const nonlands = picked.cards.WUBRG.filter((c) => !cardType(c).toLowerCase().includes('land'));
-  const totalProbability = nonlands.reduce((acc, c) => acc + probabilities[c.cardID], 0);
-  return (
-    // We can't just do pickedInCombination since that'll be empty
-    // sometimes and more than 1 step from being non-empty
-    (getColor(nonlands, picked, probabilities, cards) + getInternalSynergy(pickedInCombination, picked, cards)) *
-    totalProbability
+const calculateScore = (extendedBotState) => {
+  const { cardIndex, picked, cards, probabilities } = extendedBotState;
+  const drafting = !!cardIndex || cardIndex === 0;
+  const oracleResults = ORACLES.filter(({ draftingOnly }) => !draftingOnly || drafting).map(
+    ({ title, tooltip, computeWeight, computeValue }) => ({
+      title,
+      tooltip,
+      weight: computeWeight(extendedBotState),
+      value: computeValue(extendedBotState),
+    }),
   );
+  const score = oracleResults.reduce((acc, { weight, value }) => acc + weight * value, 0);
+  if (drafting) return { score, oracleResults };
+  const totalProbability = picked
+    .map((cIdx) => cards[cIdx])
+    .filter((c) => !cardType(c).toLowerCase().includes('land'))
+    .reduce((acc, c) => acc + probabilities[cardName(c)], 0);
+  return { score: totalProbability * score, oracleResults, totalProbability };
 };
 
-export const botRatingAndCombination = (cards, card, picked, seen, initialState, inPack = 1, packNum = 1) => {
-  const pickNum = (initialState?.[0]?.[packNum - 1]?.length ?? 0) - inPack + 1;
-  const weightedRatingScore = card ? getRating(card, cards) * getRatingWeight(packNum, pickNum, initialState) : 0;
+export const evaluateCardOrPool = (cardIndex, drafterState) => {
+  const { cards } = drafterState;
   let prevLands = { W: 0, U: 0, B: 0, R: 0, G: 0 };
-  // TODO: Only count dual lands in the combination we want to play.
-  let currentLands = { ...picked.lands };
-  let currentCombination = getCombinationForLands(currentLands);
-  let currentRating = calculateRating(
-    currentLands,
-    currentCombination,
-    card,
-    cards,
-    weightedRatingScore,
-    picked,
-    seen,
-    packNum,
-    pickNum,
-    initialState,
-  );
+  let currentLands = getLandsForPicked(drafterState);
```

Comment:
Also consider `cardIndex` since it may be a land. Probably just change the signature so we can call as `currentLands = getInitialLandsForPool((cardIndex || cardIndex === 0) ? [...drafterState.picked, cardIndex] : drafterState.picked, cards)`